test/test_invalid_headers.py

```diff
@@ -239,20 +241,33 @@ class TestFilter(object):
     ]
 
     hdr_validation_combos = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, True),
-            (True, False),
-            (False, True),
-            (False, False)
-        ]
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in (
+            itertools.product([True, False], repeat=3)
+        )
     ]
 
     hdr_validation_no_trailers = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, False),
-            (False, False)
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in [
+            (True, False, False),
+            (False, False, False),
+        ]
+    ]
```

Comment:
Sorry, I responded somewhat hastily here without reading @Lukasa's comment clearly. @Lukasa this has the same issue that I hit in #305, were testing on specifically request headers we end up with a lot of combinations that don't work. Part of me kind of feels like this test doesn't belong in the filter class, but I didn't know where to move it while working on #305.

We'll have to continually keep tweaking the hdr_validation_no_trailers list specifically for `test_matching_authority_host_headers` as we add more flags, which become fairly tedious as complexity grows. This specific test will never be a trailer, or a response. I'm also thinking it can probably only be a client too, but I'm not sure on that. Does it make sense for a server to send a request in HTTP/2?